### PR TITLE
do not ignore package-lock.json

### PR DESCRIPTION
**Pull request by jtnord**
the `package-lock.json` is vital for the build reproducibility.  Without this what is used for the build is non deterministic (withouth actually hard locking all dependencies in `package.json` which is not the case here)

this removes the ignore and checks in a package-lock.json that was created from building a clean checkout

see https://github.com/jenkinsci/bom/pull/2121#issuecomment-1571752616

amends #219 

### Submitter checklist

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by jtnord** [comment]
> This introduces changes in the local workspace just by building the project:

because something was just released - these *should* cause a CI build to fail.  will look at the project setup

**Comment by jtnord** [comment]
The Build was using `npm install` unconditionally which does not honour the `package-lock` file and hence allows builds that are not repeatable.

Switched to `npm ci` which will use the exact versions of the dependencies

see https://docs.npmjs.com/cli/v9/commands/npm-ci and https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json  

this will make the build slower for devs as `node_modules` will be wiped out - it is left as an excersize for someone else to choose if this is not wanranted and local builds should behave differently to CI by introducing a profile, however as release builds are not performed in CI this has currently not been done and I do not beleive it should not be done unless release builds also use `npm ci`  or releases are performed only in a I context (where a ci profile would be active).

**Comment by jtnord** [comment]
>  I now pinned all versions in the package.json

not if the `package-lock.json` has changed when you build the exact same sources that I did :)

**Comment by scherler** [comment]
The `ci` command is basically a `install` which whips out the `npm folder` before installing the dependencies. It will fail in certain conditions https://docs.npmjs.com/cli/v9/commands/npm-ci 
- The project must have an existing package-lock.json or npm-shrinkwrap.json.
- If dependencies in the package lock do not match those in package.json, npm ci will exit with an error, instead of updating the package lock.
- npm ci can only install entire projects at a time: individual dependencies cannot be added with this command.
- If a node_modules is already present, it will be automatically removed before npm ci begins its install.
- It will never write to package.json or any of the package-locks: installs are essentially frozen.

Basically the command is to make sure all versions of dependencies are the same as in the `lock` that makes sure that if you forget to commit the updated `lock` after you changed the `package.json` that the build will fail. It's designed to run on CI since in local dev you normally do not want your npm folder to be whipped every time.

**Comment by uhafner** [comment]
Seems to work as expected with the new changes in pom.xml now.
